fix(iota-config): [cherry-pick] remove `iota-grpc-api` dependency in `iota-config` and `iota-swarm-config`

muXxer opened the pull request:
# Description of change

This is a backport of #8859

We introduced the dependency to `iota-grpc-api` in `iota-config`, which causes that everything that depends on `iota-config` also need to install `protoc`, for example the `iota-sdk`. This PR removes this dependency.

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

thibault-martinez reviewed (approved):
Approving `crates/iota/src/iota_commands.rs`

muXxer committed:
fix(iota-config): remove `iota-grpc-api` dependency in `iota-config` and `iota-swarm-config` (#8859)

# Description of change

We introduced the dependency to `iota-grpc-api` in `iota-config`, which
causes that everything that depends on `iota-config` also need to
install `protoc`, for example the `iota-sdk`. This PR removes this
dependency.

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have checked that new and existing unit tests pass locally with
my changes
